src/jaegertracing/utils/HttpSender.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2019, The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JAEGERTRACING_UTILS_UDPCLIENT5_H
+#define JAEGERTRACING_UTILS_UDPCLIENT5_H
+
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <system_error>
+
+#include "jaegertracing/Compilers.h"
+
+#include "jaegertracing/utils/Sender.h"
+
+#include "jaegertracing/net/IPAddress.h"
+#include "jaegertracing/net/Socket.h"
+#include "jaegertracing/net/http/Response.h"
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/THttpClient.h>
+
+namespace jaegertracing {
+namespace utils {
+
+class HttpSender : public Sender {
+  public:
+    HttpSender(const net::URI& endpoint, int maxPacketSize);
+
+    ~HttpSender() { close(); }
+
+    void emitBatch(const thrift::Batch& batch) override
+    {
+        // Resets the buffer to write a new batch
+        _buffer->resetBuffer();
+
+        // Does the serialisation to Thrift
+        auto oprot = _protocol.get();
+        batch.write(oprot);
+        oprot->writeMessageEnd();
+        oprot->getTransport()->writeEnd();
+        oprot->getTransport()->flush();
+
+        uint8_t* data = nullptr;
+        uint32_t size = 0;
+        _buffer->getBuffer(&data, &size);
+        if (static_cast<int>(size) > _maxPacketSize) {

```

Comment:
a) how is _maxPacketSize defined for HTTP?
b) does it matter if it's exceeded?

Also, I haven't looked into Sender implementation here, but in other clients (e.g. Go), senders simply don't reach the state where a buffer is overflowed and spans must be dropped, because sender would flush previous spans if `append(span)` exceeds msg size. But again, this only makes sense for UDP because of fragmentation, it's not very relevant to HTTP (i.e. +/- extra span, who cares)

Reply:
it does not really matter.

I took the max from the Java [implementation](https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/HttpSender.java).

The Go impl does not have this limit. I ll remove it.